src/functions.py

```diff
@@ -71,22 +62,35 @@ def breedArch(nn1, nn2):
     Returns:
         [newcoef, newinter]: [List of coefs_ and intercepts_]
     """
-    coef1 = nn1.coefs_
-    coef2 = nn2.coefs_
-    inter1 = nn1.intercepts_
-    inter2 = nn2.intercepts_
-    newcoef = []
-    newinter = []
-
-    for i in range(min(len(coef1), len(coef2))):
-        if random.random() >= 0.5:
-            newcoef.append(coef1[i])
-        else:
-            newcoef.append(coef2[i])
-
-        if random.random() >= 0.5:
-            newinter.append(inter1[i])
-        else:
-            newinter.append(inter2[i])
-
-    return newcoef, newinter
+    layer = random.randint(0, 1)
+    shape = nn2.coefs_[layer].shape
+
+    coefFlat1 = np.ravel(nn1.coefs_[layer])
+    coefFlat2 = np.ravel(nn2.coefs_[layer])
+
+    indexes = sorted([int(random.random() * len(coefFlat1)),
+                      int(random.random() * len(coefFlat1))])
+
+    coefFlat2[indexes[0]:indexes[1]] = coefFlat1[indexes[0]:indexes[1]]
+
+    newCoefs = []
+    newCoefs.insert(layer, np.array(coefFlat2).reshape(shape))
+    newCoefs.insert(1 - layer, nn2.coefs_[1 - layer])
```

Comment:
Maybe implement a function for flatteing weights and biases as well as restoring weights and biases as list. This seems like task that will be done several times later in the project.